src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResponseResource.java

```diff
@@ -92,23 +77,76 @@ public ResponseEntity<Void> removeLockFromComplaint(@PathVariable long complaint
     }
 
     /**
-     * PUT /complaint-responses/complaint/:complaintId/resolve: resolve a complaint by updating the complaint and the associated empty complaint response
+     * PUT /complaints/{complaintId}/response: resolve a complaint by updating the complaint and the associated empty complaint response
      *
-     * @param complaintId       - id of the complaint to resolve
-     * @param complaintResponse the complaint response used for resolving the complaint
-     * @return the ResponseEntity with status 200 (Ok) and with body the complaint response used for resolving the complaint
+     * @param complaintId             - id of the complaint to resolve
+     * @param complaintResponseUpdate the complaint response used for resolving the complaint
+     * @return if action is REFRESH_LOCK: status 201 (Created) and with body the empty complaint response
+     *         if action is RESOLVE_COMPLAINT: the ResponseEntity with status 200 (Ok) and with body the complaint response used for resolving the complaint
      */
-    @PutMapping("complaint-responses/complaint/{complaintId}/resolve")
+    @PatchMapping("complaints/{complaintId}/response")
```

Comment:
@beyzaaltuntas and I discussed this extensively, we believe the patch semantics is better here, because one attribute of the response is changed here, not the whole response. Of course, this is arguable, but implementing too many "actions" or "method calls" as POST does not help either and only makes the API more complex to understand and kind of implements a RPC architecture, not a REST architecture